include/revng/Yield/Internal/Function.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <limits>
+#include <string>
+
+#include "revng/ADT/SortedVector.h"
+#include "revng/Model/VerifyHelper.h"
+#include "revng/Support/MetaAddress.h"
+#include "revng/Support/MetaAddress/YAMLTraits.h"
+#include "revng/Yield/Internal/BasicBlock.h"
+
+/* TUPLE-TREE-YAML
+
+name: Function
+type: struct
+fields:
+  - name: Entry
+    type: MetaAddress
+  - name: Name
+    type: std::string
```

Comment:
If possible, we should avoid copying names in intermediate artifacts, since it needlessly increase the situations under which we need to recompute this.

tl;dr Can you fetch this from the model later down the pipeline?

Reply:
The only issue with that is that it'd need to depend on the model in the later stages as well (as of now they are self-contained, as in they don't need anything but this intermediate structure).

But, now that you say it, yeah that would make a lot of sense! There's no reason to invalidate the entirety of this intermediate structure just because user renamed a function.

What's the safest way to store a reference to a model function? `TupleReference`? Are there problems if I were to serialize those outside the model? As in, model could change unexpectedly between serializing and deserializing those.

Reply:
I feel like not depending on the model is a bit of a misguided goal.
The safest thing is to use the `Entry` address to make a lookup on the model and get the Right Name™.

Reply:
Feels like that should work.